src/pages/studyView/StudyViewPageStore.ts

```diff
@@ -568,7 +568,8 @@ export class StudyViewPageStore {
     @autobind
     @action onCheckGene(hugoGeneSymbol: string) {
         //only update geneQueryStr whenever a table gene is clicked.
-        this.geneQueryStr = updateGeneQuery(this.geneQueries, hugoGeneSymbol);
+        this.geneQueries = updateGeneQuery(this.geneQueries, hugoGeneSymbol);
+        this.geneQueryStr = this.geneQueries.map(query=>unparseOQLQueryLine(query)).join(' ');
```

Comment:
seems like geneQueryStr is just a computation of geneQueries.  if true, then i think it could just be a @computed?

Reply:
I would just leave geneQueryStr as an observer. It's an intermediate value to deal with the geneQuery string from the GeneSelctionBox  and gene selection from the MtuatedGenes/CNA Genes table.